Ginger/Ginger/Variables/EditDatabasePage.xaml.cs

```diff
@@ -0,0 +1,157 @@
+﻿using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common;
+using GingerCore;
+using GingerCore.Actions;
+using GingerCore.Environments;
+using GingerCore.GeneralLib;
+using Microsoft.VisualStudio.Services.Common;
+using System;
+using System.Globalization;
+using System.Windows.Controls;
+using static GingerCore.Environments.Database;
+
+namespace Ginger.Variables
+{
+    /// <summary>
+    /// Interaction logic for EditDatabasePage.xaml
+    /// </summary>
+    public partial class EditDatabasePage : Page
+    {
+        private readonly Database database;
+        public EditDatabasePage(Database database, Context context)
+        {
+            this.database = database;
+            InitializeComponent();
+
+            xConnectionStringInfo.ToolTip = database.GetConnectionStringToolTip();
+
+
+            xDatabaseUserName.Init(context, database, nameof(Database.User));
+            xDatabasePassword.Init(context, database, nameof(Database.Pass));
+            xDatabaseTNS.Init(context, database, nameof(Database.TNS));
+            xDBAccEndPoint.Init(context, database, nameof(Database.User));
+            xDBAccKey.Init(context, database, nameof(Database.Pass));
+            xDatabaseConnectionString.Init(context, database, nameof(Database.ConnectionString));
+
+            ((DatabaseOperations)this.database.DatabaseOperations).NameBeforeEdit = this.database.Name;
+
+            BindingHandler.ObjFieldBinding(xDatabaseName, TextBox.TextProperty, database, nameof(Database.Name));
+            BindingHandler.ObjFieldBinding(xDatabaseDescription, TextBox.TextProperty, database, nameof(Database.Description));
+            BindingHandler.ObjFieldBinding(xKeepConnectOpen, CheckBox.IsCheckedProperty, database, nameof(Database.KeepConnectionOpen));
+            BindingHandler.ObjFieldBinding(xDatabaseType, TextBox.TextProperty, database, nameof(Database.DBType));
+
+            xDatabaseName.AddValidationRule(new DBNameValidationRule());
+
+            xDatabaseName.TextChanged += XDatabaseName_TextChanged;
+
+            if (database.DBType.Equals(eDBTypes.CosmosDb))
+            {
+                xCosmosDetailsPanel.Visibility = System.Windows.Visibility.Visible;
+                xDatabaseDetailsPanel.Visibility = System.Windows.Visibility.Collapsed;
+            }
+            else
+            {
+                xCosmosDetailsPanel.Visibility = System.Windows.Visibility.Collapsed;
+                xDatabaseDetailsPanel.Visibility = System.Windows.Visibility.Visible;
+                ChangeTheTNSName();
+                ShowBrowseBtn();
+            }
+
+        }
+
+        private void XDatabaseName_TextChanged(object sender, TextChangedEventArgs e)
+        {
+
+            string NameBeforeEdit = ((DatabaseOperations)database.DatabaseOperations).NameBeforeEdit;
```

Comment:
make sure name is changed at all placed only when user is done changing the name, this function should not get called for each character typed

Reply:
Ok, will do so